app/models/web_search.rb

```diff
@@ -48,46 +48,52 @@ def domains_scope_options
   def handle_response(response)
     @total = response.total rescue 0
     available_search_engine_pages = (@total/@per_page.to_f).ceil
-
-    if backfill_needed? and social_image_feeds_checked?
-      odie_search = initialize_odie_search(available_search_engine_pages)
-      odie_response = run_odie_search_and_handle_response(odie_search, available_search_engine_pages)
-      if odie_response && odie_response.total.zero? && (suggestion = odie_response.spelling_suggestion)
-        odie_search = initialize_odie_search(available_search_engine_pages, suggestion)
-        run_odie_search_and_handle_response(odie_search, available_search_engine_pages)
-        assign_spelling_suggestion_if_eligible(suggestion) if @indexed_results
-      end
-    end
-
+    backfill_with_odie(available_search_engine_pages) if backfill_needed? && social_image_feeds_checked?
     handle_search_engine_response(response) if available_search_engine_pages >= @page
     assign_module_tag
   end
 
+  def backfill_with_odie(available_search_engine_pages)
+    odie_search = initialize_odie_search(available_search_engine_pages)
+    odie_response = run_odie_search_and_handle_response(odie_search, available_search_engine_pages)
+    return unless odie_response&.total&.zero? && (suggestion = odie_response.spelling_suggestion)
+
+    odie_search_with_spelling_suggestions(available_search_engine_pages, suggestion)
+  end
+
   def initialize_odie_search(available_search_engine_pages, query = nil)
-    odie_search_params = @options.merge(per_page: self.default_per_page,
+    @per_page = OdieSearch.default_per_page
```

Comment:
Primary code change to ensure everything from here on out uses the OdieSearch pagination (20).

Reply:
To keep this consistent with L74, this could be written as `@per_page = odie_search_class.default_per_page`, or we could strip out the `odie_search_class` method entirely. What do you think?

Reply:
I like dumping the odie_search_class entirely.  I'm sure there was probably a reason it was like that in the first place, but that reason is probably long gone now.  Updated.